unistd/sys.c

```diff
@@ -60,6 +61,22 @@ static int shebang(const char *path)
 	return fd;
 }
 
+static int checkelf(const char *path)
+{
+	int fd;
+	unsigned char head[4];
+	unsigned char elfhead[4] = { 0x7F, 'E', 'L', 'F' };
+
+	if ((fd = open(path, O_RDONLY)) < 0)
+		return -1;
+
+	if ((read(fd, head, 4) < 0) || memcmp(head, elfhead, 4)) {
+		close(fd);
+		return -1;
+	}
+	return 0;
```

Comment:
```suggestion
	if ((read(fd, head, 4) < 4) || memcmp(head, elfhead, 4)) {
		close(fd);
		return -1;
	}
	close(fd);
	return 0;
```
Missing `close(fd);`
You should check how many bytes were actually read.

Reply:
Thank you! Added missing `close()`